### PR TITLE
feat(loginForm): login form with react+redux+router strategy

### PR DESCRIPTION
**Pull request by mariocao**
# PR Prelude

Thank you for contributing to sheikah! :)

**Please fill in this template (by putting an `x` inside
the brackets) _before_ filing your PR:**

- [x] I have read and understood [CODE_OF_CONDUCT][code] document.
- [x] I have read and understood [CONTRIBUTING][cont] document.
- [x] I have read and understood [STYLEGUIDE][style] document.
- [x] I have included tests for the changes in my PR. If not, I have included a
  rationale for why I haven't.
- [x] My code is formatted and is accepted by `yarn fmt`.
- [x] **I understand my PR may be closed if it becomes obvious I didn't
  actually perform all of these steps.**

# Why this change is necessary and useful
This PR adds the login form with the latest react+redux+router strategy.

The login is envisioned as a two step form:
1) the user selects a wallet (or requests the creation of a new wallet)
2) the user inserts the wallet password and Sheikah tries to unlock the wallet

Additionally:
+ Tests for the wallet reducer have been included
+ API renderer function to get wallet now handles errors in an elegant way :smirk: 
+ LoginForm container contains the business logic, connects to redux store and renders the ui components
+ Each step is using a separate route
+ Other ui components have been implemented for the aforementioned login form
  - Spinner
  - LoadingModal
  - SidebarLayout
  - AlertMessage

[code]: https://github.com/witnet/sheikah/blob/master/.github/CODE_OF_CONDUCT.md
[cont]: https://github.com/witnet/sheikah/blob/master/.github/CONTRIBUTING.md
[style]: https://github.com/witnet/sheikah/blob/master/docs/STYLEGUIDE.md

### CODE REVIEW

**Comment by aesedepece** [comment]
Just triggered a rebuild in one of the travis environments (nodejs v9). The other two passed :+1:

File: app/renderer/actions/loginForm.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import * as api from "app/renderer/api"
+import { Wallet } from "app/common/runtimeTypes/storage/wallets"
+import { Services } from "app/renderer/services"
+import {
+  SaveWalletAction,
+} from "./actionTypes"
+import { GetWalletResponse } from "app/common/runtimeTypes/ipc/wallets"
+
+// Function to assert a never value
+const assertNever = (x: never) => undefined
```
[the remark this answers is not in the document]
Comment (mariocao) [reply app/renderer/actions/loginForm.ts]: We saw in the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void) that `void` operator needs an expression.

File: app/renderer/ui/containers/LoginForm.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+  private walletSelectNext = async (id: string) => {
+    // Wait for the id to be set in the state
+    await this.changeState({ id })
+
+    // Dispatch action to go to next route
+    this.props.goTo(PATHS.WALLET_PASSWORD_REQUEST)
+  }
+
+  /**
+   * Method to go to the next step in WalletSelection step
```
Comment (aesedepece) [inline app/renderer/ui/containers/LoginForm.tsx]: Comment repetition with line `96`

File: app/renderer/ui/containers/LoginForm.tsx [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    // Dispatch action to go back to previous route
+    this.props.goBack()
+  }
+
+  /**
+   * Method to go to next step in WalletPasswordRequest step
+   * @param password
+   */
+  private walletPasswordRequestNext = async (password: string) => {
+    // Dispatch action to go to next route
```
Comment (aesedepece) [inline app/renderer/ui/containers/LoginForm.tsx]: This comment no longer holds true

File: app/renderer/routes.tsx [hunk 4]
```diff
@@ -17,11 +20,22 @@ export default (props: RoutesProps) => {
...
     <App>
-      <Switch>
-        <GuardedRoute path="/" guard={ifWallets(store)} aCompo={MainPage} bCompo={SignupPage} />
-        <GuardedRoute path="/main" guard={ifWallets(store)} aCompo={MainPage} bCompo={SignupPage} />
-        <Route path="/wallets/new" component={SignupPage} />
-      </Switch>
+      <RedirectedRoute
+        path="/"
+        guard={ifWallet(store)}
+        aCompo={MainPage}
```
[the remark this answers is not in the document]
Comment (mariocao) [reply app/renderer/routes.tsx]: you are right! :)

File: app/renderer/ui/containers/LoginForm.tsx (hunk 2 above)
Comment (mariocao) [reply app/renderer/ui/containers/LoginForm.tsx]: ups :sweat_smile:

**Comment by aesedepece** [comment]
Ready to merge!

**Pull request merged**